catroid/src/main/java/org/catrobat/catroid/content/actions/SetLookByIndexAction.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.actions;
+
+import android.util.Log;
+
+import org.catrobat.catroid.content.BackgroundWaitHandler;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.InterpretationException;
+
+public class SetLookByIndexAction extends SetLookAction {
+
+	private Formula formula;
+
+	@Override
+	protected void doLookUpdate() {
+
+		updateLookFromFormula();
+		if (wait) {
+			BackgroundWaitHandler.addObserver(look, this);
```

Comment:
should this really be done before the check below?
what happens if the observer gets added, but the `setLookData()` never gets called?